code/datums/antagonists/revolution.dm

```diff
@@ -0,0 +1,226 @@
+	set_antag_hud(M, null)
+
+/datum/antagonist/rev/proc/can_be_converted(mob/living/candidate)
+	if(!candidate.mind)
+		return FALSE
+	if(!can_be_owned(candidate))
+		return FALSE
+	var/mob/living/carbon/C = candidate //Check to see if the potential rev is implanted
+	if(!istype(C)) //Can't convert simple animals
+		return FALSE
+	if(C.isloyal())
+		return FALSE
+
+/datum/antagonist/rev/proc/add_revolutionary(datum/mind/rev_mind,stun = TRUE)
+	if(!can_be_converted(rev_mind.current))
+		return FALSE
+	if(stun)
+		if(iscarbon(rev_mind.current))
+			var/mob/living/carbon/carbon_mob = rev_mind.current
+			carbon_mob.silent = max(carbon_mob.silent, 5)
+			carbon_mob.flash_act(1, 1)
+		rev_mind.current.Stun(100)
+	rev_mind.add_antag_datum(/datum/antagonist/rev,rev_team)
+	rev_mind.special_role = "Revolutionary"
+	return TRUE
+
+/datum/antagonist/rev/head/proc/demote()
+	var/datum/mind/old_owner = owner
+	var/old_team = rev_team
+	silent = TRUE
+	owner.remove_antag_datum(/datum/antagonist/rev/head)
+	var/datum/antagonist/rev/new_rev = new /datum/antagonist/rev(old_owner)
+	new_rev.silent = TRUE
+	old_owner.add_antag_datum(new_rev,old_team)
+	new_rev.silent = FALSE
+	to_chat(old_owner, "<span class='userdanger'>Revolution has been disappointed of your leader traits! You are a regular revolutionary now!</span>")
+				
+/datum/antagonist/rev/farewell()
+	if(ishuman(owner.current))
+		owner.current.visible_message("[owner.current] looks like they just remembered their real allegiance!", \
+				"<span class='userdanger'><FONT size = 3>You are no longer a brainwashed revolutionary! Your memory is hazy from the time you were a rebel...the only thing you remember is the name of the one who brainwashed you...</FONT></span>")
+	else if(issilicon(owner.current))
+		owner.current.visible_message("The frame beeps contentedly, purging the hostile memory engram from the MMI before initalizing it.", \
+				"<span class='userdanger'><FONT size = 3>The frame's firmware detects and deletes your neural reprogramming! You remember nothing but the name of the one who flashed you.</FONT></span>")
+
+/datum/antagonist/rev/proc/remove_revolutionary(borged, deconverter)
+	log_attack("[owner.current] (Key: [key_name(owner.current)]) has been deconverted from the revolution by [deconverter] (Key: [key_name(deconverter)])!")
+	if(borged)
+		message_admins("[ADMIN_LOOKUPFLW(owner.current)] has been borged while being a [name]")
+	owner.special_role = null
+	if(iscarbon(owner.current))
+		var/mob/living/carbon/C = owner.current
+		C.Unconscious(100)
+	owner.remove_antag_datum(type)
+
+/datum/antagonist/rev/head/remove_revolutionary(borged,deconverter)
+	if(!borged)
+		return
+	. = ..()
+
+/datum/antagonist/rev/head/equip_rev()
+	var/mob/living/carbon/human/H = owner.current
+	if(!istype(H))
+		return
+
+	if (owner.assigned_role == "Clown")
+		to_chat(owner, "Your training has allowed you to overcome your clownish nature, allowing you to wield weapons without harming yourself.")
```

Comment:
nitpick #2: This fluff won't line up with a clown being promoted to head rev later in the round